lib/awful/titlebar.lua

```diff
@@ -782,6 +845,13 @@ function titlebar.widget.button(c, name, selector, action)
             if state ~= "" then
                 state = "_" .. state
             end
+            -- try select user defined tooltip texts according to state
+            local tooltip_msgs = beautiful["titlebar_tooltip_messages"] or default_tooltip_messages
```

Comment:
Ok, I get it : `beautiful.titlebar_tooltip_messages` is a table, we have to make sure it exists before trying to access the actual string values.

(Hm, that's honestly something that reinforce my idea that we shouldn't use a table for titlebar_tooltip_messages.)

(If we want to keep this table as is) Could we at least have a util function `get_tooltip_msg(name, img)` to handle these checks and "default waterfall logic"?  I don't see any point where we shouldn't have to do these checks to get the tooltip message, and it would make the code more explicit/self-explaining.